cpp/src/interop/to_arrow_host.cu

```diff
@@ -147,7 +147,7 @@ int dispatch_to_arrow_host::operator()<bool>(ArrowArray* out) const
   NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), NANOARROW_TYPE_BOOL, column));
 
   NANOARROW_RETURN_NOT_OK(populate_validity_bitmap(ArrowArrayValidityBitmap(tmp.get())));
-  auto bitmask = bools_to_mask(column, stream, mr);
+  auto bitmask = detail::bools_to_mask(column, stream, mr);
```

Comment:
why is `detail::` added here? 
There are no namespace changes of this API in this PR.

Reply:
For some APIs, we only distinguished between the detail and public function by the inclusion of a stream parameter. Adding the stream parameter makes the resolution ambiguous. Therefore, we need to add a detail namespace.